pkg/microservice/user/core/handler/user/user_group.go

```diff
@@ -0,0 +1,224 @@
+/*
+Copyright 2023 The KodeRover Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package user
+
+import (
+	"fmt"
+
+	"github.com/gin-gonic/gin"
+	"github.com/koderover/zadig/pkg/microservice/user/core/service/user"
+
+	internalhandler "github.com/koderover/zadig/pkg/shared/handler"
+	e "github.com/koderover/zadig/pkg/tool/errors"
+)
+
+type createUserGroupReq struct {
+	Name        string   `json:"name"`
+	Description string   `json:"description"`
+	UIDs        []string `json:"uids"`
+}
+
+func CreateUserGroup(c *gin.Context) {
+	ctx := internalhandler.NewContext(c)
+	defer func() { internalhandler.JSONResponse(c, ctx) }()
+
+	// this is local, so we simply generate user auth info from service
+	err := GenerateUserAuthInfo(ctx)
+	if err != nil {
+		ctx.UnAuthorized = true
+		ctx.Err = fmt.Errorf("failed to generate user authorization info, error: %s", err)
+		return
+	}
+
+	// user needs to be an admin to create user groups
+	if !ctx.Resources.IsSystemAdmin {
+		ctx.Logger.Errorf("user %s is not system admin, cannot create user groups", ctx.UserID)
+		ctx.UnAuthorized = true
+		return
+	}
+
+	req := new(createUserGroupReq)
+	err = c.BindJSON(req)
+	if err != nil {
+		ctx.Err = e.ErrInvalidParam
+		return
+	}
+
+	ctx.Err = user.CreateUserGroup(req.Name, req.Description, req.UIDs, ctx.Logger)
```

Comment:
Seems like it don't have operation logs for user group

Reply:
not designed to